src/masternode-payments.cpp

```diff
@@ -615,8 +615,12 @@ bool CMasternodePayments::GetBlockTxOuts(int nBlockHeight, CAmount blockReward,
         } else {
             CAmount operatorReward = (masternodeReward * dmnPayee->nOperatorReward) / 10000;
             masternodeReward -= operatorReward;
-            voutMasternodePaymentsRet.emplace_back(masternodeReward, dmnPayee->pdmnState->scriptPayout);
-            voutMasternodePaymentsRet.emplace_back(operatorReward, dmnPayee->pdmnState->scriptOperatorPayout);
+            if (masternodeReward > 0) {
+                voutMasternodePaymentsRet.emplace_back(masternodeReward, dmnPayee->pdmnState->scriptPayout);
+            }
+            if (operatorReward > 0) {
```

Comment:
This is not needed, we already know that operator reward is `>0` (because of `if` at line 613). But if you want to generalize it, you could probably rewrite the whole thing (line 613+) the other way around i.e. calculate rewards first and then `emplace_back` them if conditions (amount and/or script) are ok.

Reply:
My reasoning here was that rewards will eventually drop to a point where the operator reward calculation will also result in 0, even though a != 0 percentage was given. But generalizing it sounds good, would make things easier to understand.